TAScheduler/UnitTests/tests.py

```diff
@@ -76,71 +77,879 @@ class TestAdminCourseInstrAsgmt(TestCase):  # Kevin
 class TestAdminCourseTAAsgmt(TestCase):  # Kevin
     pass
 
-class TestTAInit(TestCase):
+
+class TestTAInit(TestCase):  # Kevin
     pass
 
+
+# I ORIGINALLY INTERPRETTED:
+# "instructor max assignments" as max "course" assignments for an instructor &
+# "ta max assignments" as max "lab" assignments for a TA ... I think it might be better to just make it "max course assignments" for both?
 class TestTAHasMaxAsgmts(TestCase):  # Kiran
-    pass
+    taDB = None
+    courseDB = None
+    user = None  # for TA
+    taObj = None
+    adminObj = None  # for deleting course
+
+    def setUp(self):
+        self.user = User.objects.create(
+            email_address='TA@example.com',
+            password='TApassword',
+            first_name='TA',
+            last_name='User',
+            home_address='123 TA St',
+            phone_number='1234567890'
+        )
+        self.taDB = TA.objects.create(user=self.user, max_assignments=1)  # max 1 assignment!
+        self.taObj = TAObj(self.taDB)  # creating TA object using TA in database.
+        tempUserForAdmin = User(email_address="admin@example.com")
+        tempAdmin = Administrator(user=tempUserForAdmin)
+        self.adminObj = AdminObj(tempAdmin)
+
+        self.courseDB = Course.objects.create(
+            course_id=101,
+            semester='Fall 2023',
+            name='Introduction to Testing',
+            description='A course about writing tests in Django.',
+            num_of_sections=3,
+            modality='Online',
+            credits=4
+        )
+
+    # 1] TA w/ 1 course assignment
+    def test_1Crse1MaxCap(self):
+        TAToCourse.objects.create(ta=self.taDB, course=self.courseDB)  # assigning TA to course using db?
+        self.assertEquals(self.taObj.hasMaxAsgmts(), True,
+                          msg="TA has 1 max assignments & assigned 1 course: @ max cap")
+
+    # 2] TA w/ 0 course assignment
+    def test_0Crse1MaxCap(self):
+        self.assertEquals(self.taObj.hasMaxAsgmts(), False,
+                          msg="TA has 1 max assignments & not assigned 1 course: not @ max cap")
+
+    # 3] TA w/ max cap -> no max assign
+    def test_origMaxCapToNoAsgn(self):
+        TAToCourse.objects.create(ta=self.taDB, course=self.courseDB)
+        self.adminObj.removeCourse(CourseObj(self.courseDB))  # removing course should also remove this TA's assignment
```

Comment:
Do not have methods from the views_methods.py file used in your test except the method you are testing. This test could fail because of a poorly coded .removeCourse method.

Reply:
I agree - I feel like these are great tests to ensure that both methods are working correctly.

Dr. Boyland included tests like this in 351 (I know it's an entirely different course) so I don't think it's completely unreasonable to ensure 2 methods are consistent with each other.

Also, I asked TA for the Fraction HW if we could do this, he said that it was fine.
Exameple:
Having a test to ensure that (.add(4,4)) = (.mult(4,2)) , 4+4 = 4*2 is a great way to test 2 methods in 1 unitTest